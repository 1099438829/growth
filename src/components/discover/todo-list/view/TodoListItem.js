import React, { Component, PropTypes } from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import { Icon } from 'react-native-elements';
import Line from '../../../Line';

class TodoListItem extends Component {
  static componentName = 'TodoListItem';

  static propTypes = {
    checked: PropTypes.bool,
    title: PropTypes.string.isRequired,
    click: PropTypes.func,
  };

  static defaultProps = {
    checked: false,
    title: '',
    click: () => {},
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this.props.click}
        underlayColor={'rgba(250, 250, 250, .9)'}
      >
        <View style={{ flexDirection: 'row', height: 40, paddingLeft: 10 }}>
          <Icon
            name={this.props.checked
              ? 'md-checkbox'
              : 'md-square-outline'}
            type={'ionicon'}
            color={this.props.checked ? '#03a9f4' : null}
          />
          <View style={{ flex: 1, justifyContent: 'center', marginLeft: 25 }}>
            <Text style={{ lineHeight: 39 }} numberOfLines={1}>{this.props.title}</Text>
            <Line />
          </View>
        </View>
      </TouchableHighlight>
    );
  }
}
export default TodoListItem;
