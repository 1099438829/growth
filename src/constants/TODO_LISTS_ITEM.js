import TODO_LISTS from './TODO_LISTS';

const TODO_LIST = TODO_LISTS['zh-cn'];
const ITEM = [
  TODO_LIST.hello,
  TODO_LIST.before,
  TODO_LIST.frontend,
  TODO_LIST.backend,
  TODO_LIST.env,
  TODO_LIST.test,
  TODO_LIST.refactor,
  TODO_LIST.configurable,
  TODO_LIST.ci,
  TODO_LIST.container,
  TODO_LIST.retro,
  TODO_LIST.server,
  TODO_LIST.analytics,
  TODO_LIST.arch,

];
export default ITEM;
