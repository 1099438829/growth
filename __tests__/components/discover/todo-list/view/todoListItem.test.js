import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import TodoListItem from '../../../../../src/components/discover/todo-list/view/TodoListItem';
import TODO_LISTS_ITEM from '../../../../../src/constants/TODO_LISTS_ITEM';

it('renders correctly', () => {
  const data = TODO_LISTS_ITEM[0].basic[0];
  const todoListItem = renderer.create(
    <TodoListItem
      title={data.title}
      click={() => {}}
      checked
    />,
  );
  const todoListItemJson = todoListItem.toJSON();
  expect(todoListItemJson).toMatchSnapshot();
});
