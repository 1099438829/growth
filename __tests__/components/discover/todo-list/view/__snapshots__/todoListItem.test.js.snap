// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "paddingLeft": 10,
      }
    }
  >
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#03a9f4",
              "fontSize": 24,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 25,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "lineHeight": 39,
          }
        }
      >
        用该语言写hello,world
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
